lisa/wa.py

```diff
@@ -598,16 +609,28 @@ def trace_idle_analysis(trace):
     NAME = 'trace'
     _ARTIFACT_NAME = 'trace-cmd-bin'
     _PURE_GET_JOB_DF = False
+    _FORCE_AUTO = True
 
-    def __init__(self, wa_output, trace_to_df, **kwargs):
+    def __init__(self, wa_output, trace_to_df=None, **kwargs):
         self._trace_to_df = trace_to_df
         self._trace_kwargs = kwargs
         super().__init__(wa_output, df_postprocess=None)
 
     def _get_artifact_df(self, path):
+        if self._trace_to_df is None:
+            raise ValueError('trace_to_df was not specified for the TraceCollector')
         trace = Trace(path, **self._trace_kwargs)
         return self._trace_to_df(trace)
 
+    @property
+    @memoized
+    def traces(self):
+        return {
```

Comment:
In the name of saving some computations, I'd use a lazy mapping object. You can easily make one by inheriting from `collections.abc.Mapping` and implementing the relevant methods. This should go in `lisa.utils`. I can look like:
```
lazy = LazyMapping({'key1': (1, 2), 'key2': (3, 4)}, lambda x1, x2: x1 + x2)
dict(lazy) == {'key1': 3, 'key2': 7}
```